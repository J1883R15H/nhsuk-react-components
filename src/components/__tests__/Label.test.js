/* eslint-disable jsx-a11y/label-has-associated-control */
import React from 'react';
import { shallow } from 'enzyme';
import Label from '../label';

describe('Label', () => {
  it('matches snapshot', () => {
    const shallowMount = shallow(<Label>Label</Label>);
    expect(shallowMount).toMatchSnapshot();

    shallowMount.unmount();
  });
  it('renders as bold', () => {
    const shallowMount = shallow(<Label type="bold">Label</Label>);
    expect(
      shallowMount.containsMatchingElement(
        <label className="nhsuk-label  nhsuk-label--s ">Label</label>
      )
    ).toBeTruthy();
    shallowMount.unmount();
  });
  it('renders as a heading', () => {
    const shallowMount = shallow(<Label type="heading">Label</Label>);
    expect(
      shallowMount.containsMatchingElement(
        <label className="nhsuk-label   nhsuk-label--xl">Label</label>
      )
    ).toBeTruthy();
    shallowMount.unmount();
  });
});
