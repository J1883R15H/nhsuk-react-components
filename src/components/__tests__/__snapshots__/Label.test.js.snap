// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label matches snapshot 1`] = `
<label
  className="nhsuk-label"
  htmlFor=""
  id="__label"
  style={Object {}}
>
  Label
</label>
`;
